### PR TITLE
feat: add CQRS boilerplate

### PR DESCRIPTION
**Pull request by kubabialy**
This pull request makes the following changes:
- Adding CQRS boilerplate for future development.

Test checklist:
- [x] Unit tests added and passing
- [x] Example added

- [x] I certify that I ran my checklist

Ping @ushahidi/platform

### CODE REVIEW

File: app/Providers/AppServiceProvider.php [hunk 1]
```diff
@@ -195,4 +202,24 @@ public function registerFeatures()
             return new Features($app[ConfigRepository::class]);
         });
     }
+
+    private function registerBusses(): void
+    {
+        $this->app->singleton(CommandBus::class, function ($app) {
+            $commandBus = new CommandBus($app);
+
+            $commandBus->register(ExampleCommand::class, ExampleCommandHandler::class);
```
Comment (webong) [inline app/Providers/AppServiceProvider.php:211]: I am considering we have an automated way to register actions to the bus, maybe specified in a config file.

```
// config/cqrs.php
<?php
return [
   'queries' => [
        \Ushahidi\App\Bus\Query\Example\ExampleQuery::class => \Ushahidi\App\Bus\Query\Example\ExampleQueryHandler::class,
 ],
 
  'commands' => [
        \Ushahidi\App\Bus\Command\Example\ExampleCommand::class => \Ushahidi\App\Bus\Command\Example\ExampleCommandHandler::class,
 ],
];
```

Then in the `registerBusses` function in service provider we have this to register queries (vice versa for commands)

```
foreach (config('cqrs.queries') as $action => $handler) {
        $queryBus->register($action, $handler);
 }
```

**Comment by webong** [comment]
@kubabialy looks good to me 👍🏽

File: app/Providers/AppServiceProvider.php (hunk 1 above)
Comment (kubabialy) [reply app/Providers/AppServiceProvider.php:211]: I don't think it's any more automated than what's in there really. But adds on complexity as there more files to keep track of.

**Review by webong — APPROVED** [review]

File: app/Providers/AppServiceProvider.php (hunk 1 above)
Comment (webong) [reply app/Providers/AppServiceProvider.php:211]: In the scenario, we have about 10 - 20 commands and queries to register, one would have to register in the AppServiceProvider making it a large file to work with, this is what I look at avoiding. Also, we only have one file `cqrs.php` in the config folder, where any one can go and register an action to the bus.

About complexity, I really don't think so ... this is a pattern widely adopted in the laravel ecosystem i.e the `app.php` uses such approach in registering providers.

Comment (Mh-Asmi) [reply app/Providers/AppServiceProvider.php:211]: I agree with @webong about this point,  this file might be a quite large after adding many commands
& queries

**Review by Mh-Asmi — APPROVED** [review]

**Pull request merged**